editions/test/tiddlers/tests/test-filters.js

```diff
@@ -467,6 +467,18 @@ function runTests(wiki) {
 		// Ensure that join correctly handles empty strings in source
 		expect(wiki.filterTiddlers("[[]] Paul +[join[-]]").join(",")).toBe("-Paul");
 		expect(wiki.filterTiddlers("[[ John ]] [[Paul ]] [[ George]] Ringo +[trim[]join[-]]").join(",")).toBe("John-Paul-George-Ringo");
+		expect(wiki.filterTiddlers("[[ abc ]] [[def ]] [[ ghi]] +[trim[]]").join(",")).toBe("abc,def,ghi");
+		expect(wiki.filterTiddlers("[[ abc ]] [[def ]] [[ ghi]] +[trim:prefix[]]").join(",")).toBe("abc ,def ,ghi");
+		expect(wiki.filterTiddlers("[[ abc ]] [[def ]] [[ ghi]] +[trim:suffix[]]").join(",")).toBe(" abc,def, ghi");
+		expect(wiki.filterTiddlers("[[ abacus ]] [[ baobab ]] +[trim[ab]]").join(",")).toBe(" abacus , baobab ");
+		expect(wiki.filterTiddlers("[[ abacus ]] [[ baobab ]] +[trim[a]]").join(",")).toBe(" abacus , baobab ");
```

Comment:
/*disapointed*/ .. But then there is no difference between removeprefix and removesuffix. .. I think `trim` should _always_ remove whitespace + eventual chars.

Reply:
But that's just my opinion. ... What do others think?

Reply:
The difference between `trim` and `removePrefix` would be that `removePrefix` also *filters out* any titles that don't start with the string, whereas `trim` does no filtering. That was the original feature request: `removePrefix` without filtering, so the fact that trim-with-parameter acts exactly like `removePrefix` except for the filtering is precisely the design goal I'm going for here.

If I implement it this way, then what you want (`trim` removes whitespace and then the supplied characters) can easily be achieved by doing `+trim[]trim[a]`. If I implement it the way you're suggesting, then what I want (exactly like `removePrefix` but without filtering) would *not* be possible, since I would have no way to tell it *not* to remove any whitespace that I wanted to keep. Therefore, it's better to implement this the more flexible way, which allows for both use cases; and that's the way I wrote it here.

Reply:
`+trim[]trim[a]` you are right. That's much more verbose and will create less support requests. I'm OK thx.